tests/foreman/ui/test_variables.py

```diff
@@ -180,336 +451,398 @@ def test_negative_update_type(self):
+        with Session(self.browser) as session:
+            make_smart_variable(
+                session,
+                name=name,
+                default_value=initial_value,
+                puppet_class=self.puppet['name'],
+            )
+            for data in invalid_sc_variable_data():
+                with self.subTest(data):
+                    self.smart_variable.update(
+                        name,
+                        key_type=data['sc_type'],
+                        default_value=data['value'],
+                    )
+                    self.assertIsNotNone(
+                        self.smart_variable.wait_until_element(
+                            common_locators['haserror'])
+                    )
+                    self.smart_variable.click(self.smart_variable.search(name))
+                    value = self.smart_variable.wait_until_element(
+                        locators['smart_variable.default_value']).text
+                    self.assertEqual(value, initial_value)
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_negative_validate_default_value_required_checkbox(self):
-        """Error is raised for blank default Value - Required checkbox.
-
-        @id: cdbe5feb-daf1-49c4-98c0-5bbbd436c3e8
-
-        @steps:
-
-        1.  Create a variable with empty value.
-        2.  Check Required checkbox in 'Optional Input Validator'.
-        3.  Submit the change.
-
-        @assert: Error is raised for blank default value by 'Required'
-        checkbox.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_validate_default_value_required_checkbox(self):
-        """Error is not raised for default Value - Required checkbox.
-
-        @id: 09e42df9-2fe1-444f-a603-6f7b6b3f127e
-
-        @steps:
-
-        1.  Provide some default value.
-        2.  Check Required checkbox in 'Optional Input Validator'.
-        3.  Submit the change.
-
-        @assert: Error is not raised default value by 'Required' checkbox.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_negative_validate_matcher_value_required_checkbox(self):
-        """Error is raised for blank matcher Value - Required checkbox.
-
-        @id: 7e02efb6-e1a2-4872-877a-78075b2de182
-
-        @steps:
-
-        1.  Create a matcher for variable for some attribute.
-        2.  Dont provide Value for matcher. Keep blank.
-        3.  Check Required checkbox in 'Optional Input Validator'.
-        4.  Submit the change.
-
-        @assert: Error is raised for blank matcher value by 'Required'
-        checkbox.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_validate_matcher_value_required_checkbox(self):
```

Comment:
`Required` checkbox functionality is not supported by Smart Variables and actually it is not even present on UI